src/overmap_location.cpp

```diff
@@ -28,11 +29,21 @@ const overmap_location &string_id<overmap_location>::obj() const
 bool overmap_location::test( const int_id<oter_t> &oter ) const
 {
     return std::any_of( terrains.cbegin(), terrains.cend(),
-    [ &oter ]( const string_id<oter_type_t> &type ) {
+    [ &oter ]( const oter_type_str_id & type ) {
         return oter->type_is( type );
     } );
 }
 
+const oter_type_id overmap_location::get_random_terrain() const
+{
+    return random_entry( terrains );
+}
+
+const std::string overmap_location::get_random_terrain_string() const
```

Comment:
Why this? The caller can just do the `.str()` call directly (`get_random_terrain().str()`).

Reply:
It seems it works like this:
a
```CPP
std::string dest_type = get_location_for( needs.front() )->get_random_terrain().id().str();
```